### PR TITLE
Add the feature of selecting text from left margin

### PR DESCRIPTION
**Pull request by AfraaAlobaid**
- [ YES] I documented all behaviour as far as I could do.
  - [YES ] I target the develop-branch, and *not* the master branch.
  - [ YES] I have tested this extensively and paid extra attention to
        potential cross-platform issues (e.g. Cmd/Ctrl/Super-key bindings)
  - [NO ] I have made use of ESLint using the provided configuration from the
        repository's .eslintrc.json file, and it did not complain.
  - [NO ] I have added an entry to the CHANGELOG.md.
  - [YES ] I matched my code-style to the repository (as far as possible).
  - [ YES] I do agree that my code will be published under the GNU GPL v3 license.
  - [ YES] As far as JS-files are concerned, I made sure to copy (in case of new files)
        or adapt (in case of existing files) the info in the header.
  - [YES ] I synced the latest commits to develop shortly before proposing
        so that no merge issues occur.

## Description
We have added the feature that enables the user to select text by pressing the left mouse button on the left margin. 

## Changes
1. Whenever the user presses the LEFT mouse button (you need a check for that), you can set some variable in the data-property of the Editor.vue-component that contains the start line where the user has pressed the left mouse button.
2. When the user then MOVES the mouse (that event is not yet implemented, but you should be able to easily do so) while the left mouse button is pressed (e.g. after the mousedown-event has fired && the button was in fact the left mouse button), you need to select all lines from the start line to the current cursor position.
3. When the user RELEASES the left mouse button, stop adding any lines to the current selection and unset that start variable.
4. When the user then presses the left mouse button AGAIN, set that start line variable again and do the process anew. Please remember that you should REPLACE any existing selection in this case.

## Additional information
Tested on: 

Ubuntu 18.04.5
Windows 10
macOS Big Sur v 11.3.1

### CODE REVIEW

File: source/win-main/Editor.vue
Comment (nathanlesage) [inline source/win-main/Editor.vue:605]: That's a very good catch, and from the looks of it precisely what would work!

File: source/win-main/Editor.vue
Comment (nathanlesage) [inline source/win-main/Editor.vue:597]: You can merge this if-statement with the one above (or do a check for event.button !== 0 and return) — that would improve readability and reduce the chance that any changes afterwards end up accidentally AFTER the closing bracket!

File: source/win-main/Editor.vue
Comment (nathanlesage) [inline source/win-main/Editor.vue:600]: You can convert these to `const` (improves readability as it is clear those variables won't be changed)

Comment (nathanlesage) [reply source/win-main/Editor.vue:600]: Ah, and: That startpoint should rather be called anchor (because that makes it clear that this is the anchor of the whole selection even as it is being resized) AND also saved as a variable of the Vue component (that is, set this in `data` above).

The reason is that you can then factor out the custom mousemove-event you are listening to after mousedown has been pressed. Inside the mousedown-event (which will be triggered every time) you can then make a check that the anchor is actually set. So here you should set it to the given coordinates as you do, and on mouseup you should set it to `undefined` or `null`. In the mousemove-handler (which should be implemented just like the mousedown and mouseup) you can then at the beginning just check for `anchor !== undefined`

File: source/win-main/Editor.vue
Comment (nathanlesage) [inline source/win-main/Editor.vue:611]: As indicated above, please convert this into a vue-event (`v-on:mousemove`)

File: source/win-main/Editor.vue
Comment (nathanlesage) [inline source/win-main/Editor.vue:631]: Good catch!

File: source/win-main/Editor.vue
Comment (nathanlesage) [inline source/win-main/Editor.vue:640]: As stated above: Instead of removing the event listener, set the anchor to null here

**Review by nathanlesage — CHANGES_REQUESTED** [review]
Very good work, you've even caught two things I didn't anticipate. Basically there's only one last thing to do, that is, move the event listener for mousemoves into its own dedicated vue-event, and then I'll test and merge. But it does look very good so far!

**Comment by AfraaAlobaid** [comment]
Thank you very much for the quick feedback. In the last commit the anchor has been saved to data in the vue component and the mouse move event is changed to a vue event and handled in the same fashion as other events.

File: source/win-main/Editor.vue
Comment (nathanlesage) [inline source/win-main/Editor.vue:615]: Here it's safe to perform a strict check (`===`).

**Comment by nathanlesage** [comment]
Looks almost perfect, there are just some code styling issues – after function names or "if"s must come a space before the left bracket, and the curly brackets should also be a little bit spaced. Do you have ESLint in use? It should automatically flag those things so that you can adapt these and then the PR looks ready to test & merge.

**Comment by AfraaAlobaid** [comment]
Thank you Hendrik, now it should be styled properly.

**Comment by sayoniroy** [comment]
@nathanlesage Hi Hendrik, is there anything else that we need to do regarding this issue? We were wondering if we could get an update. Thanks 😊

File: source/win-main/Editor.vue
Comment (nathanlesage) [inline source/win-main/Editor.vue:606]: It's better to make this a `const`

**Review by nathanlesage — COMMENTED** [review]
This looks wonderful! I'll be testing out this PR during the next week, and then it's ready for merging :)

**Comment by nathanlesage** [comment]
I just tested this PR, and it works like a charm! Thank you very much. As soon as you are happy, I will merge it!

**Comment by nathanlesage** [comment]
Excuse my confusion, I just realised there's no more need for changes! Merging …

**Pull request merged**
